### PR TITLE
Better external auth integration, restore + site button, tests!

### PR DESCRIPTION
**Pull request by pmusaraj**
Various fixes:
- catch external auth requests and open them using https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession (better UI for users, previously they could be lost)
- catches regular login requests too, fixing issues with passkey logins: https://meta.discourse.org/t/discourse-hub-login-with-passkey-completely-broken/328441/4 
- adds back the + site button for direct access to adding sites by URL, see https://meta.discourse.org/t/can-t-add-new-forum-to-discoursehub/320026/15?u=pmusaraj
- fixes the sharing extension (regressed in iOS 18)
- fixes navigation in non-English locales
- reload last URL on webview crash, a partial fix for https://meta.discourse.org/t/discoursehub-on-ipad-is-closing-unexpectedly/327190
- updates app icon

Plus, some basic e2e tests have been added!

### CODE REVIEW

File: ios/Discourse/Images.xcassets/AppIcon.appiconset/discourse.png
Comment (pmusaraj) [inline ios/Discourse/Images.xcassets/AppIcon.appiconset/discourse.png:1]: At last, we can give iOS only 1 icon file and have it handle all sizes on its own!

File: js/Discourse.js [hunk 1]
```diff
@@ -561,8 +547,9 @@ class Discourse extends React.Component {
                     tabBarBackground: () => this._blurView(theme.name),
                   }}>
                   <Tab.Screen
-                    name={i18n.t('home')}
+                    name="Home"
```
Comment (markvanlan) [inline js/Discourse.js:550]: I know this was intentional but it's ok to lose i18n?

Comment (markvanlan) [reply js/Discourse.js:550]: Oh wow I'm an idiot. I see it now.

**Review by markvanlan — APPROVED** [review]
All the logic looks good to me! I'm not pulling it down and testing it -- you said it has been running for a bit for a while on several devices.

File: js/Discourse.js (hunk 1 above)
Comment (pmusaraj) [reply js/Discourse.js:550]: Yeah, this was a pretty stupid regression on my part. It caused some ugly routing errors for users on non-English locales. (That's why there are tests for this now.)

**Comment by pmusaraj** [comment]
Thanks @markvanlan!

**Pull request merged**